projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/launchpad-save-modal.php

```diff
@@ -22,8 +22,9 @@ function add_launchpad_options() {
 	}
 
 	$launchpad_options = array(
-		'launchpadScreenOption' => get_option( 'launchpad_screen' ),
-		'siteIntentOption'      => get_option( 'site_intent' ),
+		'launchpadScreenOption'       => get_option( 'launchpad_screen' ),
+		'siteIntentOption'            => get_option( 'site_intent' ),
```

Comment:
Given that Launchpad is new, and site intent has been around a while, I'm surprised that we have a way to retrieve launchpad options, but there isn't a way to simply retrieve site intent without needing to do this round-a-about method of passing it through jetpack options.

Reply:
> I'm surprised that we have a way to retrieve launchpad options, but there isn't a way to simply retrieve site intent without needing to do this round-a-about method of passing it through jetpack options.

Ah do you mean we should be retrieving this information from an endpoint? I agree with you, although hydrating initial state like this is a convention I've seen in other areas of Jetpack ( can't remember exactly where )

Reply:
>do you mean we should be retrieving this information from an endpoint

I was just saying I'm surprised it's not already available - via the same kind of front end hydration that jetpack is apparently doing for launchpad options. Site intent has been around and central to our sites much longer than launchpad.

Reply:
I'm not sure I understand what you're saying, because site intent options _are_ being made available through the same front end hydration process?

Reply:
Just as a note, there will probably be a dedicated endpoint for most of this data by the time we're done resolving Atomic-related issues. Also there's nothing wrong with injecting the values in this way.